src/librustc_trans/diagnostics.rs

```diff
@@ -12,6 +12,96 @@
 
 register_long_diagnostics! {
 
+E0510: r##"
+`return_address` was used in an invalid context. Erroneous code example:
+
+```
+extern "rust-intrinsic" {
+    fn return_address() -> *const u8;
+}
+
+pub unsafe fn by_value() -> i32 {
+    let _ = return_address();
+    // error: invalid use of `return_address` intrinsic: function does
+    //        not use out pointer
+    0
+}
+```
+
+Returned values are stored in registers. In the case where the returned
+type doesn't fit in a register, the function returns `()` and has an
+additional input argument, this is a pointer where the result should
+be written. Example:
+
+```
+extern "rust-intrinsic" {
+    fn return_address() -> *const u8;
+}
+
+pub unsafe fn by_pointer() -> String {
+    let _ = return_address();
+    String::new() // ok!
+}
+```
+"##,
+
+E0511: r##"
+An invalid monomorphization on an intrinsic function was used.
```

Comment:
“An intrinsic was monomorphized invalidly.” maybe? You can’t “use” monomorphization “on” something.